lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -119,14 +113,9 @@ sub results {
     for my $step (@$details) {
         my $text_file_name = $step->{text};
         if (!$skip_text_data && $text_file_name && !defined $step->{text_data}) {
-            eval { $step->{text_data} = decode('UTF-8', path($dir, $text_file_name)->slurp); };
-            if (my $error = $@) {
-                # try reading the results one more time if the JSON file's size has increased; otherwise render an error
-                # note: Likely a concurrent finalize_job_results Minion job has finished so the separate text file has
-                #       just been incorporated within the JSON file.
-                return $self->results(%options) if (-s $file_name // -1) > ($initial_file_size // -1);
-                $step->{text_data} = "Unable to read $text_file_name.";
-            }
+            my $text_file = path($dir, $text_file_name);
+            my $text_data = -r $text_file ? decode('UTF-8', $text_file->slurp) : undef;
```

Comment:
The readable check that was missing in the previous version of the patch. It's needed because the text file might not have been uploaded yet.